### PR TITLE
DDF-1366 Re-implemented the SourcePoller

### PR DESCRIPTION
**Pull request by emmberk**
##### ABBREVIATED REVIEW BETWEEN 2.13.X AND MASTER IS IN EFFECT
Links to 2.13.x PRs:
- https://github.com/codice/ddf/pull/3858
- https://github.com/codice/ddf/pull/4407
____

#### What does this PR do?
This PR ports over the changes for the new Source Poller implementation. Additionally, this PR includes a new commit to move the Source Poller code from core-api-impl to standard-framework because it is only used in the Standard Framework. It is not whitelisted.

_The file changes in this PR do not need to be re-reviewed, but this PR does need to be re-heroed._

#### Who is reviewing it? 
@jhunzik @Kjames5269 @lessarderic @mcalcote @pklinef @vinamartin

#### Select relevant component teams: 
@codice/core-apis 
@codice/io 

#### Ask 2 committers to review/merge the PR and tag them here.
@brjeter
@clockard

#### How should this be tested?
See the test steps on https://github.com/codice/ddf/pull/3858 and https://github.com/codice/ddf/pull/4407. Additionally, confirm that the Source Poller does not cause any issues (searching, federating, log messages, source availability) when the DDF is a registry or is connected to a registry.
```log:set DEBUG org.codice.ddf.catalog.sourcepoller```

#### What are the relevant tickets?
[DDF-1366](https://codice.atlassian.net/browse/DDF-1366)

#### Screenshots
image

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by emmberk** [comment]
build now

**Comment by clockard** [comment]
@emmberk Thanks for moving this out of catalog-core-api-impl!

**Comment by emmberk** [comment]
`mvn clean install` from a clean .m2 succeeded for me locally

**Comment by emmberk** [comment]
Not sure why the last build just stopped: https://jenkins.phx.connexta.com/service/jenkins/job/DDF-Jobs/job/pr/job/Linux/6392/

build now

**Comment by emmberk** [comment]
❗️This PR is waiting on the hero. DO NOT MERGE.

File: catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/SourceOperations.java [hunk 1]
```diff
@@ -345,41 +356,49 @@ private SourceInfoResponse getFanoutSourceInfo(SourceInfoRequest sourceInfoReque
       }
 
       // Fanout will only add one source descriptor with all the contents
-      Set<Source> availableSources =
+      // Using a List here instead of a Set because we should not rely on how Sources are compared
```
[the remark this answers is not in the document]
Comment (emmberk) [reply catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/SourceOperations.java]: I didn't change this logic in the `SourceOperations` class. I didn't change the `frameworkProperties.getFederatedSources()` part. `QueryOperations` and SourceOperations have different responsibilities. `QueryOperations` handles querying the specified sources, which could include the local store. `SourceOperations` is used to return responses for the /services/catalog/sources endpoint, which doesn't need to local store.

**Comment by emmberk** [comment]
rebased
build now

**Comment by ahoffer** [comment]
Hero successful. I installed a pre-built distro provided by @emmberk. 
Set logging to debug the source poller. `log:set DEBUG org.codice.ddf.catalog.sourcepoller`
Added loopback federation sources and adjusted the polling interval.

System behaved as designed. From the Intrigue UI, sources that were actually available, but had never been pinged by the poller displayed as unavailable -- this is expected because the poller did not know about them. However, because they were actually available, they did return results that could be seen from Intrigue. Again, this is working as designed.

@emmberk and I performed other source poller test, both over federated systems and with loopback federation.

**Pull request merged**